packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -143,13 +143,19 @@ const ThirdPartySignInForm = ({
   const { logViewEventOnce } = useMetrics();
   const stateRef = useRef<HTMLInputElement>(null);
 
-  function onClick() {
+  async function onClick() {
     logViewEventOnce(`flow.${party}`, 'oauth-start');
 
-    if (viewName === 'signin' && party === 'google') {
-      GleanMetrics.thirdPartyAuth.startGoogleAuthFromLogin();
-    } else if (viewName === 'signin' && party === 'apple') {
-      GleanMetrics.thirdPartyAuth.startAppleAuthFromLogin();
+    switch (`${party}-${viewName}`) {
+      case 'google-signin':
+        await GleanMetrics.thirdPartyAuth.startGoogleAuthFromLogin();
```

Comment:
Why do we want to await on these metrics now?

Reply:
Since the page will unload as the user will be taken to the third party authorization page, I want to ensure the event is sent prior to that.  However, that's not the way to go about it.  I'll make a change.  Thank you for the question!

Reply:
Cool, I'd be interested to know what is the way to go about it - I've had to add metrics to this flow before and I've seen them go through during debug testing, but now I'm not sure if there could be async bugs that you mention.